packages/react-component-library/src/components/NumberInput/Input.tsx

```diff
@@ -29,10 +33,41 @@ export const Input: React.FC<InputProps> = ({
   label,
   size,
   value,
+  onChange,
   ...rest
 }) => {
   const hasLabel = !!(label && label.length)
 
+  const KEY_ARROW_UP = 'ArrowUp'
+  const KEY_ARROW_DOWN = 'ArrowDown'
+
+  const handleKeyDown: KeyboardEventHandler<HTMLInputElement> = (event) => {
```

Comment:
An option to consider is whether this could live inside its own hook such as `useInputKeys`.